src/api/java/mezz/jei/api/recipe/category/IRecipeCategory.java

```diff
@@ -36,9 +37,22 @@
 	/**
 	 * Returns the localized name for this recipe type.
 	 * Drawn at the top of the recipe GUI pages for this category.
+	 * @deprecated Prefer calling and using {@link #getTitleAsTextComponent()}.
 	 */
+	@Deprecated
 	String getTitle();
 
+	/**
+	 * Returns a text component representing the name of this recipe type.
+	 * Drawn at the top of the recipe GUI pages for this category.
+	 * @since JEI 7.6.4
+	 */
+	default ITextComponent getTitleAsTextComponent() {
+		//TODO: When getTitle is removed this should just be rename to getTitle and not be defaulted anymore.
+		String title = getTitle();
+		return title == null ? null : new StringTextComponent(title);
```

Comment:
The recipe category tab has a null check on getTitle so I put one here as well, though I am not sure if it should be here as well or not.

Reply:
The null check there is to help with broken plugins or when devs are just starting the plugin.
This is good for now, it will no longer be exposed in the API after `getTitle` is removed.